### PR TITLE
Added Support for Returning Attention Scores in TransformerEncoder call

### PR DESCRIPTION
**Pull request by anirudhr20**
**Summary:** This pull request introduces a new feature that adds support for optionally returning attention scores in the `TransformerEncoder` class. This is controlled by the `return_attention_scores` flag, which when set to `True`, returns both the output and the attention scores from the attention mechanism.

**Changes Introduced:**
- Updated the call method in the `TransformerEncoder` to handle the `return_attention_scores` flag.
- Refactored the code to ensure that the attention scores are computed and returned when required.
- Updated the documentation in the `call` method to reflect the changes.
- Added unit tests to ensure the correctness of this feature using pytest.

**Testing:**
Ran the unit tests to verify that the `return_attention_scores` flag works as expected.
`return_attention_scores=True` (verifies that attention scores are returned and the shapes are correct).
Related Issue: https://github.com/keras-team/keras-hub/issues/1644

### CODE REVIEW

**Review by SamanehSaadat — CHANGES_REQUESTED** [review]
Thanks, @anirudhr20! The PR looks great! I just left a couple of nit comments in the test.

File: keras_hub/src/layers/modeling/transformer_encoder_test.py [hunk 1]
```diff
@@ -109,3 +109,14 @@ def test_mask_propagation(self):
...
         outputs = encoder(inputs)
         self.assertAllEqual(outputs._keras_mask, mask)
+
+    def test_attention_scores(self):
+        encoder = TransformerEncoder(intermediate_dim=4, num_heads=2)
+        inputs = random.uniform(shape=[1, 4, 6])
+        outputs, attention_scores = encoder(
+            inputs, return_attention_scores=True
+        )
+        print(attention_scores)
```
[the remark this answers is not in the document]
Comment (anirudhr20) [reply keras_hub/src/layers/modeling/transformer_encoder_test.py:119]: Yes will remove the print statement. Thanks for pointing it out!

File: keras_hub/src/layers/modeling/transformer_encoder_test.py [hunk 2]
```diff
@@ -109,3 +109,14 @@ def test_mask_propagation(self):
...
+        assert outputs.shape == inputs.shape
+        # attention scores shape (batch_size, num_of_attn_heads, seq_length, seq_length)
+        assert attention_scores.shape == [1, 2, 4, 4]
```
[the remark this answers is not in the document]
Comment (anirudhr20) [reply keras_hub/src/layers/modeling/transformer_encoder_test.py]: Thanks I have made the changes

[the remark this answers is not in the document]
Comment (SamanehSaadat) [reply keras_hub/src/layers/modeling/transformer_encoder_test.py]: Thanks for making the changes!

**Review by SamanehSaadat — APPROVED** [review]
Thanks for the PR!

**Pull request merged**